docker-cmd-mac.sh

```diff
@@ -0,0 +1,168 @@
+    echo "  EXEC: $EXEC [-e, --exec], CONTAINER: \"$CONTAINER\" (use docker ps for the id)"
+    echo "  COMMAND: \"$COMMAND\" [--cmd]"
+    echo "  EXP: \"$EXP\" [--exp]"
+    echo "  DETACH: ($DETACH) [--detach]"
+    echo "  DEVELOPMENT dir \"$DEVEL_DIR\" [-d, --devel]"
+    echo "  LEARNING dir \"$LEARN_DIR\" [-l, --learn]"
+}
+
+for arg in "$@"
+do
+    case $arg in ## -l=*|--lib=*) DIR="${i#*=}" is the way to retrieve the parameter
+        -n=*|--image-name=*)
+        IMAGE_NAME="${arg#*=}"
+        ;;
+        -f=*|--image-folder=*)
+        IMAGE_FOLDER="${arg#*=}"
+        ;;
+        --no-gpu)
+        GPU=false
+        GPU_OPT=""
+        ;;
+        --with-optirun)
+        OPTIRUN=true
+        OPTIRUN_OPT="optirun"
+        ;;
+        --detach)
+        DETACH=true
+        ;;
+        -b|--build)
+        BUILD=true
+        ;;
+        -c|--cache) # it does nothing without BUILD=true
+        CACHE=true
+        ;;
+        -r|--run)
+        RUN=true
+        ;;
+        -s|--screen) # it does nothing without RUN=true
+        SCREEN=true
+        ;;
+        -e=*|--exec=*) # it works only with RUN=false
+        EXEC=true
+        CONTAINER="${arg#*=}"
+        ;;
+        --cmd=*)
+        COMMAND="${arg#*=}"
+        ;;
+        --exp=*)
+        EXP="${arg#*=}"
+        ;;
+        -d=*|--devel=*)
+        DEVEL_DIR="${arg#*=}"
+        ;;
+        -l=*|--learn=*)
+        LEARN_DIR="${arg#*=}"
+        ;;
+        *)
+        # unknown option
+        echo "Unknown option \"$arg\""
+        print_help
+        exit
+        ;;
+    esac
+done
+
+print_help
+
+# Tensorflow original image
+# docker run -u $(id -u):$(id -g) --gpus all -it --rm tensorflow/tensorflow:latest-gpu-py3 bash
+
+## Building the docker image
+if [[ "$BUILD" = true ]]; then
+    if [[ "$CACHE" = true ]]; then
+        echo "Building the docker container using the cache, if present."
+        $OPTIRUN_OPT docker build \
+            --build-arg USER_ID=$(id -u ${USER}) \
+            --build-arg GROUP_ID=$(id -g ${USER}) \
+            -t "$IMAGE_NAME" "$IMAGE_FOLDER"
+    else
+        echo "Building the docker container ignoring the cache, even if present."
+        $OPTIRUN_OPT docker build \
+            --build-arg USER_ID=$(id -u ${USER}) \
+            --build-arg GROUP_ID=$(id -g ${USER}) \
+            --no-cache -t "$IMAGE_NAME" "$IMAGE_FOLDER"
+    fi
+fi
+
+if [[ "$RUN" = true ]]; then
+    # My docker build
+    MOUNT_DEVEL=""
+    if [[ $DEVEL_DIR ]]; then
+        MOUNT_DEVEL="--mount src=$DEVEL_DIR,target=/home/alice/devel,type=bind"
+    fi
+    MOUNT_LEARN=""
+    if [[ $LEARN_DIR ]]; then
+        MOUNT_LEARN="--mount src=$LEARN_DIR,target=/home/alice/learning,type=bind"
+    fi
+    CONT_NAME=""
+    if [[ $EXP ]]; then
+        CONT_NAME="--name $EXP"
+    fi
+    if [[ "$DETACH" = true ]]; then
+        DETACH="-d"
+    else
+        DETACH=""
+    fi
+    CURR_UID=$(id -u)
```

Comment:
Here the variables 
```
CURR_UID=$(id -u)
CURR_GID=$(id -g)
``` 
are still set and pass to docker, even if they are hardcoded in the Dockerfile.

Reply:
now can also be set through command line 3700450ad87758e25a10864ddf52b2d86cbad11b